compilation error in qt/masternodelist.cpp ERROR: ‘class QDateTime’ has no member named ‘offsetFromUtc’

hi there. I was trying to compile dash and the make command stopped here:

> 
>  CXX      qt/qt_libbitcoinqt_a-masternodelist.o
> qt/masternodelist.cpp: In member function ‘void MasternodeList::updateMyMasternodeInfo(QString, QString, masternode_info_t&)’:
> qt/masternodelist.cpp:193:172: **error: ‘class QDateTime’ has no member named ‘offsetFromUtc’**
>  Mn.fInfoValid ? infoMn.nTimeLastPing + QDateTime::currentDateTime().offsetFromUtc() : 0)));
>                                                                      ^~~~~~~~~~~~~
> qt/masternodelist.cpp: In member function ‘void MasternodeList::updateNodeList()’:
> qt/masternodelist.cpp:276:173: error: ‘class QDateTime’ has no member named ‘offsetFromUtc’
>  %Y-%m-%d %H:%M", mn.lastPing.sigTime + QDateTime::currentDateTime().offsetFromUtc())));
>                                                                      ^~~~~~~~~~~~~
> At global scope:
> cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
> Makefile:6850: recipe for target 'qt/qt_libbitcoinqt_a-masternodelist.o' failed
> make[2]: *** [qt/qt_libbitcoinqt_a-masternodelist.o] Error 1
> make[2]: Leaving directory '/home/d/src3/dash/src'
> Makefile:9045: recipe for target 'all-recursive' failed
> make[1]: *** [all-recursive] Error 1
> make[1]: Leaving directory '/home/d/src3/dash/src'
> Makefile:652: recipe for target 'all-recursive' failed
> make: *** [all-recursive] Error 1

Does anyone knows whats wrong?
I am using Qt version 4.8.7 in /usr/lib/i386-linux-gnu

--- a/src/qt/masternodelist.cpp
+++ b/src/qt/masternodelist.cpp
@@ -15,6 +15,17 @@
 #include <QTimer>
 #include <QMessageBox>
 
+int GetOffsetFromUtc()
+{
+#if QT_VERSION < 0x050200
+    const QDateTime dateTime1 = QDateTime::currentDateTime();
+    const QDateTime dateTime2 = QDateTime(dateTime1.date(), dateTime1.time(), Qt::UTC);
+    return dateTime1.secsTo(dateTime2);
+#else
+    return QDateTime::currentDateTime().offsetFromUtc();
+#endif
+}
+
 MasternodeList::MasternodeList(const PlatformStyle *platformStyle, QWidget *parent) :
     QWidget(parent),
     ui(new Ui::MasternodeList),
@@ -186,13 +197,6 @@ void MasternodeList::updateMyMasternodeInfo(QString strAlias, QString strAddr, c
 
     masternode_info_t infoMn;
     bool fFound = mnodeman.GetMasternodeInfo(outpoint, infoMn);
-#if QT_VERSION < 0x050200
-    const QDateTime dateTime1 = QDateTime::currentDateTime();
-    const QDateTime dateTime2 = QDateTime(dateTime1.date(), dateTime1.time(), Qt::UTC);
-    int offsetFromUtc = dateTime1.secsTo(dateTime2);
-#else
-    int offsetFromUtc = QDateTime::currentDateTime().offsetFromUtc();
-#endif
 
     QTableWidgetItem *aliasItem = new QTableWidgetItem(strAlias);
     QTableWidgetItem *addrItem = new QTableWidgetItem(fFound ? QString::fromStdString(infoMn.addr.ToString()) : strAddr);
@@ -200,7 +204,7 @@ void MasternodeList::updateMyMasternodeInfo(QString strAlias, QString strAddr, c
     QTableWidgetItem *statusItem = new QTableWidgetItem(QString::fromStdString(fFound ? CMasternode::StateToString(infoMn.nActiveState) : "MISSING"));
     QTableWidgetItem *activeSecondsItem = new QTableWidgetItem(QString::fromStdString(DurationToDHMS(fFound ? (infoMn.nTimeLastPing - infoMn.sigTime) : 0)));
     QTableWidgetItem *lastSeenItem = new QTableWidgetItem(QString::fromStdString(DateTimeStrFormat("%Y-%m-%d %H:%M",
-                                                                                                   fFound ? infoMn.nTimeLastPing + offsetFromUtc : 0)));
+                                                                                                   fFound ? infoMn.nTimeLastPing + GetOffsetFromUtc() : 0)));
     QTableWidgetItem *pubkeyItem = new QTableWidgetItem(QString::fromStdString(fFound ? CBitcoinAddress(infoMn.pubKeyCollateralAddress.GetID()).ToString() : ""));
 
     ui->tableWidgetMyMasternodes->setItem(nNewRow, 0, aliasItem);
@@ -270,13 +274,7 @@ void MasternodeList::updateNodeList()
     ui->tableWidgetMasternodes->clearContents();
     ui->tableWidgetMasternodes->setRowCount(0);
     std::map<COutPoint, CMasternode> mapMasternodes = mnodeman.GetFullMasternodeMap();
-#if QT_VERSION < 0x050200
-    const QDateTime dateTime1 = QDateTime::currentDateTime();
-    const QDateTime dateTime2 = QDateTime(dateTime1.date(), dateTime1.time(), Qt::UTC);
-    int offsetFromUtc = dateTime1.secsTo(dateTime2);
-#else
-    int offsetFromUtc = QDateTime::currentDateTime().offsetFromUtc();
-#endif
+    int offsetFromUtc = GetOffsetFromUtc();
 
     for(auto& mnpair : mapMasternodes)
     {

--- a/src/qt/masternodelist.cpp
+++ b/src/qt/masternodelist.cpp
@@ -186,6 +186,13 @@ void MasternodeList::updateMyMasternodeInfo(QString strAlias, QString strAddr, c
 
     masternode_info_t infoMn;
     bool fFound = mnodeman.GetMasternodeInfo(outpoint, infoMn);
+#if QT_VERSION < 0x050200
+    const QDateTime dateTime1 = QDateTime::currentDateTime();
+    const QDateTime dateTime2 = QDateTime(dateTime1.date(), dateTime1.time(), Qt::UTC);
+    int offsetFromUtc = dateTime1.secsTo(dateTime2);
+#else
+    int offsetFromUtc = QDateTime::currentDateTime().offsetFromUtc();
+#endif
 
     QTableWidgetItem *aliasItem = new QTableWidgetItem(strAlias);
     QTableWidgetItem *addrItem = new QTableWidgetItem(fFound ? QString::fromStdString(infoMn.addr.ToString()) : strAddr);
@@ -193,7 +200,7 @@ void MasternodeList::updateMyMasternodeInfo(QString strAlias, QString strAddr, c
     QTableWidgetItem *statusItem = new QTableWidgetItem(QString::fromStdString(fFound ? CMasternode::StateToString(infoMn.nActiveState) : "MISSING"));
     QTableWidgetItem *activeSecondsItem = new QTableWidgetItem(QString::fromStdString(DurationToDHMS(fFound ? (infoMn.nTimeLastPing - infoMn.sigTime) : 0)));
     QTableWidgetItem *lastSeenItem = new QTableWidgetItem(QString::fromStdString(DateTimeStrFormat("%Y-%m-%d %H:%M",
-                                                                                                   fFound ? infoMn.nTimeLastPing + QDateTime::currentDateTime().offsetFromUtc() : 0)));
+                                                                                                   fFound ? infoMn.nTimeLastPing + offsetFromUtc : 0)));
     QTableWidgetItem *pubkeyItem = new QTableWidgetItem(QString::fromStdString(fFound ? CBitcoinAddress(infoMn.pubKeyCollateralAddress.GetID()).ToString() : ""));
 
     ui->tableWidgetMyMasternodes->setItem(nNewRow, 0, aliasItem);
@@ -263,6 +270,13 @@ void MasternodeList::updateNodeList()
     ui->tableWidgetMasternodes->clearContents();
     ui->tableWidgetMasternodes->setRowCount(0);
     std::map<COutPoint, CMasternode> mapMasternodes = mnodeman.GetFullMasternodeMap();
+#if QT_VERSION < 0x050200
+    const QDateTime dateTime1 = QDateTime::currentDateTime();
+    const QDateTime dateTime2 = QDateTime(dateTime1.date(), dateTime1.time(), Qt::UTC);
+    int offsetFromUtc = dateTime1.secsTo(dateTime2);
+#else
+    int offsetFromUtc = QDateTime::currentDateTime().offsetFromUtc();
+#endif
 
     for(auto& mnpair : mapMasternodes)
     {
@@ -273,7 +287,7 @@ void MasternodeList::updateNodeList()
         QTableWidgetItem *protocolItem = new QTableWidgetItem(QString::number(mn.nProtocolVersion));
         QTableWidgetItem *statusItem = new QTableWidgetItem(QString::fromStdString(mn.GetStatus()));
         QTableWidgetItem *activeSecondsItem = new QTableWidgetItem(QString::fromStdString(DurationToDHMS(mn.lastPing.sigTime - mn.sigTime)));
-        QTableWidgetItem *lastSeenItem = new QTableWidgetItem(QString::fromStdString(DateTimeStrFormat("%Y-%m-%d %H:%M", mn.lastPing.sigTime + QDateTime::currentDateTime().offsetFromUtc())));
+        QTableWidgetItem *lastSeenItem = new QTableWidgetItem(QString::fromStdString(DateTimeStrFormat("%Y-%m-%d %H:%M", mn.lastPing.sigTime + offsetFromUtc)));
         QTableWidgetItem *pubkeyItem = new QTableWidgetItem(QString::fromStdString(CBitcoinAddress(mn.pubKeyCollateralAddress.GetID()).ToString()));
 
         if (strCurrentFilter != "")

